src/librbd/ImageCtx.cc

```diff
@@ -729,14 +729,8 @@ librados::IoCtx duplicate_io_ctx(librados::IoCtx& io_ctx) {
     std::unique_lock image_locker(image_lock);
 
     // reset settings back to global defaults
-    for (auto& key : config_overrides) {
-      std::string value;
-      int r = cct->_conf.get_val(key, &value);
-      ceph_assert(r == 0);
-
-      config.set_val(key, value);
-    }
     config_overrides.clear();
+    config.set_config_values(cct->_conf.get_config_values());
```

Comment:
just wondering (again), couldn't the copy constructor be used here instead?

Reply:
Copy constructor cannot be used because `md_config_t` is non-copyable.